ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java

```diff
@@ -163,6 +169,14 @@ protected Stream<Optional<BuilderMapEntry>> createMilestones(
         create(config.getArrowGlacierBlockNumber(), specFactory.arrowGlacierDefinition(config)),
         create(config.getGrayGlacierBlockNumber(), specFactory.grayGlacierDefinition(config)),
         create(config.getMergeNetSplitBlockNumber(), specFactory.parisDefinition(config)),
+        // Timestamp Forks
+        createTimestampMilestone(config.getShanghaiTime(), specFactory.shanghaiDefinition(config)),
+        createTimestampMilestone(config.getCancunTime(), specFactory.cancunDefinition(config)),
+        createTimestampMilestone(
+            config.getFutureEipsTime(), specFactory.futureEipsDefinition(config)),
+        createTimestampMilestone(
+            config.getExperimentalEipsTime(), specFactory.experimentalEipsDefinition(config)),
+
```

Comment:
append timestamp forks to end of blockNumber forks.

Using createTimestampMilestone because the getBlockByHeader algorithm needs to know the difference between timestamp and blockNumber forks.